### PR TITLE
Use anyhow instead of the deprecated failure crate

### PR DESCRIPTION
**Pull request by romoh**
**What this PR does / why we need it**:
Use [anyhow](https://github.com/dtolnay/anyhow) for error handling instead of failure crate which is [deprecated](https://github.com/rust-lang-nursery/failure#failure---a-new-error-management-story). anyhow has parity with failure so the change is 1:1 and no major changes were required.
#89 

**Special notes for your reviewer**:
None

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [X] added code adheres to standard Rust formatting (`cargo fmt`)
- [X] code builds properly (`cargo build`)
- [X] code is free of common mistakes (`cargo clippy`)
- [X] all Akri tests succeed (`cargo test`)
- [X] inline documentation builds (`cargo doc`)
- [X] version has been updated appropriately (`./version.sh`)

### CODE REVIEW

**Review by kate-goldenring — APPROVED** [review]
Thanks for fixing this, @romoh! We are working on updating our workflow files so that the tests are no longer failing. When https://github.com/deislabs/akri/pull/202 merges, could you rebase? Then, (fingers crossed!) the tests will all be passing.

**Comment by kate-goldenring** [comment]
> Can you run `version.sh -u -p` again?

PR train has been moving in reasonably quickly. @romoh, can you update the version until its greater than [main](https://github.com/deislabs/akri/blob/main/version.txt) currently 0.1.9?

File: agent/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ blake2 = "0.8.0"
 chrono = "0.4.10"
 cfg-if = "0.1"
 env_logger = "0.6.1"
-failure = "0.1.5"
+anyhow = "1.0.38"
```
Comment (jiria) [inline agent/Cargo.toml:16]: nit: should we fix the ordering?

Comment (kate-goldenring) [reply agent/Cargo.toml:16]: good catch, @jiria. we should. would you be willing to put in the fix?

**Pull request merged**